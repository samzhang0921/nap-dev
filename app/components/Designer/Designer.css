.noSelected {}

.filterCheckbox {
  display: inline-block;
  height: 15px;
  width: 15px;
  border: 1px solid #C2C2C2;
  -webkit-border-radius: 2px;
  border-radius: 2px;
  margin-right: 6px;
}

.checkBox {
  background-color: #000;
  color: #FFF;
  text-decoration: none;
  flex-grow: 1;
}

span {
  align-self: center;
}

.designer {
    line-height: 14px;
    vertical-align: top;
    display: inline-block;
    max-width: 145px;
    margin-bottom: 11px;
}
