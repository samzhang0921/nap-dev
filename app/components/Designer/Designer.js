import React from 'react';
import {render} from 'react-dom';
import style from './Designer.css';

export default class Designer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectBox:this.props.select
    };
    this.getDesigner = this.getDesigner.bind(this);
  }
  componentWillReceiveProps(nextProps){
    console.log(nextProps);
    this.setState({
      selectBox:nextProps.select
    })
  }
  getDesigner (designer) {
    let { selectBox } = this.state;
    this.setState({
      selectBox: !selectBox
    });
    this.props.updateDesignerList(designer);
  }

  render() {
    console.log("111444", this.state.selectBox);
    const checkBox = this.state.selectBox? style.filterCheckedbox: style.filterCheckbox;
    return (
      <li data-designer-id={this.props.designerId} onClick={()=>this.getDesigner(this.props.designerId)}>
          <span className={checkBox}></span>
          <div className={style.designer}>
            <span>{this.props.designerName}</span>
          </div>
        <div></div>
      </li>
    )
  }
}
