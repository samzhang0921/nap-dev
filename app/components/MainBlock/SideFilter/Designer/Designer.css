.noSelected {}

.filterCheckbox {
  display: inline-block;
  height: 15px;
  width: 15px;
  border: 1px solid #C2C2C2;
  -webkit-border-radius: 2px;
  border-radius: 2px;
  margin-right: 4px;
  transition: transform 0.2s ease-in;
  -ms-transition: transform 0.2s ease-in;
  -moz-transition: transform 0.2s ease-in;
  -webkit-transition: transform 0.2s ease-in;
}
.filterCheckedbox {
  composes:filterCheckbox;
  background-color: #000;
  transform: rotate(360deg);

}

.checkBox {
  background-color: #000;
  color: #FFF;
  text-decoration: none;
  flex-grow: 1;
}

span {
  align-self: center;
}

.designer {
    line-height: 14px;
    vertical-align: top;
    display: inline-block;
    max-width: 145px;
    margin-bottom: 11px;
}
