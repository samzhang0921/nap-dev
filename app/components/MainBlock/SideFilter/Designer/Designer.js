import React from 'react';
import {render} from 'react-dom';
import style from './Designer.css';
import PropTypes from 'prop-types';

class Designer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectBox:false
    };
    this.getDesigner = this.getDesigner.bind(this);
  }

  componentWillUpdate(nextProps, nextState){
    if (this.props.clearClicked !== nextProps.clearClicked){
      this.setState({
        selectBox:false
      })
    }
  }

  getDesigner (designer) {
    const { selectBox } = this.state;
    this.setState({
      selectBox: !selectBox
    });
    this.props.updateDesignerList(designer);
  }

  render() {
    const checkBox = this.state.selectBox? style.filterCheckedbox: style.filterCheckbox;
    return (
      <li data-designer-id={this.props.designerId} onClick={()=>this.getDesigner(this.props.designerId)}>
          <span className={checkBox}></span>
          <div className={style.designer}>
            <span>{this.props.designerName}</span>
          </div>
        <div></div>
      </li>
    )
  }
};

Designer.propTypes = {
  designerName :PropTypes.string,
  designerId :PropTypes.number,
  updateDesignerList :PropTypes.func,
  clearClicked :PropTypes.bool
};

export default Designer;
