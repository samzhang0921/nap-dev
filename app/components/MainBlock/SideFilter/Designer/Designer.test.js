
import React from 'react';
import ReactDOM from 'react-dom';
import Designer from './Designer';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Designer', () => {
  const data = ['L', 'b', 'c'];

    it('should render a todo item ', () => {
      const tree = toJson(shallow(<Designer data={[]} />));
      expect(tree).toMatchSnapshot();
    });

    it('should match its snapshot with data', () => {
      const data = ['Learn react', 'rest', 'go out'];
      const tree = renderer.create(
        <Designer data={data} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should pass a selected value to the onChange function', () => {
   const tree = renderer.create(
     <Designer />
   );
   const component = shallow(<Designer data={data} />);
   component.find('li').first().simulate('change', { target: {
     value: 'Change function' }
   });

   expect(toJson(component)).toMatchSnapshot();
 });

});
