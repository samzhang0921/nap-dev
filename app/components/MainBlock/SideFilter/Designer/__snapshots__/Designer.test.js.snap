// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Designer should match its snapshot with data 1`] = `
<li
  data-designer-id={undefined}
  onClick={[Function]}
>
  <span
    className="filterCheckbox"
  />
  <div
    className="designer"
  >
    <span />
  </div>
  <div />
</li>
`;

exports[`Designer should pass a selected value to the onChange function 1`] = `
<li
  onClick={[Function]}
>
  <span
    className="filterCheckbox"
  />
  <div
    className="designer"
  >
    <span />
  </div>
  <div />
</li>
`;

exports[`Designer should render a todo item  1`] = `
<li
  onClick={[Function]}
>
  <span
    className="filterCheckbox"
  />
  <div
    className="designer"
  >
    <span />
  </div>
  <div />
</li>
`;
